src/main/webapp/app/course/learning-paths/services/base-api-http.service.ts

```diff
@@ -9,6 +9,23 @@ export abstract class BaseApiHttpService {
 
     private readonly baseUrl = 'api';
 
+    /**
+     * Debounces a function call to prevent it from being called multiple times in a short period.
+     * @param callback The function to debounce.
+     * @param delay The delay in milliseconds to wait before calling the function.
+     */
+    public static debounce<T extends unknown[]>(callback: (...args: T) => void, delay: number) {
```

Comment:
I really like the idea of debouncing the requests here! 😍 Did you purposely decide not to use the Angular-provided functionality for debouncing but implement it yourself? https://v17.angular.io/guide/http-optimize-server-interaction

Reply:
The idea is to gradually introduce `Promises` with `async` and `await`. The background for that is that with the introduction of signals etc. Angular slowly wants to move away from having Rxjs built in as also stated in the latest RFCs of Angular. The reason for that is that Rxjs mostly introduces complexity which is 99% of the time not really necessary and therefore leads to often wrong usage (Here is e.g. a good blog about the wrong usage of subscribe which turns async code into sync code: https://medium.com/whozapp/working-with-observables-in-angular-what-we-learned-at-whoz-55e2d26e5a03).
Long story short :) this way Angular wants to be more developer friendly especially for beginners. And that's why I introduced the base http service at all and also the debounce method.

Reply:
Okay sure but does that explain why you coded your own `debounce` function? I think Angular provides one themselves that (on the first look) does not utilize `Rxjs` too

Reply:
The debounce of Angular is just the debounce method of Rxjs. If you see a pipe() function somewhere it's always Rxjs